src/main/client/type.c

```diff
@@ -1671,35 +1668,22 @@ AerospikeClient *AerospikeClient_New(PyObject *parent, PyObject *args,
 {
     AerospikeClient *self = (AerospikeClient *)AerospikeClient_Type.tp_new(
         &AerospikeClient_Type, args, kwds);
-    as_error err;
-    as_error_init(&err);
-    int return_code = 0;
-    return_code = AerospikeClient_Type.tp_init((PyObject *)self, args, kwds);
 
-    switch (return_code) {
-    // 0 Is success
-    case 0: {
+    int return_code =
+        AerospikeClient_Type.tp_init((PyObject *)self, args, kwds);
+    if (return_code == 0) {
         return self;
     }
-    case -1: {
-        if (PyErr_Occurred()) {
-            return NULL;
-        }
-        break;
-    }
-    default: {
-        if (PyErr_Occurred()) {
-            return NULL;
-        }
-        break;
-    }
+    if (PyErr_Occurred()) {
+        goto CLEANUP;
     }
 
-    PyObject *py_err = NULL;
+    as_error err;
+    as_error_init(&err);
     as_error_update(&err, AEROSPIKE_ERR_PARAM, "Failed to construct object");
-    error_to_pyobject(&err, &py_err);
-    PyObject *exception_type = raise_exception(&err);
-    PyErr_SetObject(exception_type, py_err);
-    Py_DECREF(py_err);
+    raise_exception(&err);
+
+CLEANUP:
+    AerospikeClient_Type.tp_free(self);
```

Comment:
We're planning to move away from the C-based Python client soon, so I'm not sure if it's worth rewriting the code to use PyObject_Call. This might be more complicated to do because the client isn't normally instantiated using a class constructor; you would use `aerospike.client(config)` to create a new client object.

Reply:
I thought this would be a fairly simple replace, but I'm not 100% sure. If you think it's more trouble than its worth feel free to move ahead with what is here.